recipe/meta.yaml

```diff
@@ -1,113 +1,158 @@
+        # windows cannot link with mix debug and release version, so set to Release
+        - cmake -GNinja -DCMAKE_BUILD_TYPE=Release .  # [win]
+        - cmake --build .
 
   - name: libprotobuf-static
-    script: build_static.sh  # [unix]
-    script: bld-static.bat  # [win]
+    script: build-lib.sh  # [unix]
+    script: build-lib.bat  # [win]
+    build:
+      always_include_files:
+        # Must overwrite the CMake metadata from shared libprotobuf
+        - lib/cmake/protobuf/           # [unix]
+        - Library/lib/cmake/protobuf/   # [win]
+      ignore_run_exports_from:
+        # jsoncpp not needed at runtime
+        - jsoncpp
+        # gtest not needed at runtime
+        - gtest
     requirements:
       build:
         - {{ compiler('c') }}
         - {{ compiler('cxx') }}
-        - cmake  # [win]
-        - ninja  # [win]
-        - autoconf  # [not win]
-        - automake  # [not win]
-        - libtool  # [not win]
-        - pkg-config  # [not win]
-        - unzip  # [not win]
-        - make  # [not win]
+        - cmake
+        - ninja
       host:
-        - zlib    # [not win]
-        - {{ pin_subpackage('libprotobuf', exact=True) }}  # [not win]
+        - gtest 1.14.0
+        - jsoncpp 1.9.4
+        - abseil-cpp 20230802
+        - zlib
+        - {{ pin_subpackage('libprotobuf', exact=True) }}
       run:
-        - zlib    # [not win]
-        - {{ pin_subpackage('libprotobuf', exact=True) }}  # [not win]
-      run_constrained:
-        - libprotobuf <0a0  # [win]
+        - {{ pin_subpackage('libprotobuf', exact=True) }}
     test:
+      requires:
+        # cmake needs compiler to be able to run package detection, see
+        # https://discourse.cmake.org/t/questions-about-find-package-cli-msvc/6194
+        - {{ compiler('cxx') }}
+        - cmake
+        - ninja
+        - zlib
+        - zstd
+      files:
+        - cmake_test/
       commands:
-        - test -f ${PREFIX}/lib/libprotobuf.a  # [unix]
-        - test -f ${PREFIX}/lib/libprotobuf-lite.a  # [unix]
-        - test -f ${PREFIX}/lib/libprotoc.a  # [unix]
-        - if exist %PREFIX%\\Library\\bin\\libprotoc.dll exit 1  # [win]
-        - if not exist %PREFIX%\\Library\\lib\\libprotoc.lib exit 1  # [win]
-        - if not exist %PREFIX%\\Library\\lib\\libprotobuf.lib exit 1  # [win]
-        - if not exist %PREFIX%\\Library\\lib\\libprotobuf-lite.lib exit 1  # [win]
+        {% for each_lib in libraries %}
+        # static libraries
+        - test -f ${PREFIX}/lib/{{ each_lib }}.a                       # [unix]
+        - if not exist %LIBRARY_LIB%\{{ each_lib }}-static.lib exit 1  # [win]
+        {% endfor %}
+
+        # more CMake integration
+        - cd cmake_test
+        - cmake -GNinja $CMAKE_ARGS .                 # [unix]
+        # windows cannot link with mix debug and release version, so set to Release
+        - cmake -GNinja -DCMAKE_BUILD_TYPE=Release .  # [win]
+        - cmake --build .
```

Comment:
What is being tested here exactly?

Reply:
beside protoc executable and static/shared libraries package provides cmake definitions which could be reused while using the this package for building with upstreams, we testing here just that, see also cmake_tests/CMakeLists.txt

Reply:
can you test the validity of these CMake definitions?